### ISSUE TITLE
NSException when attempting to call Style.getLayer() on a text-based SymbolLayer

### ISSUE THREAD

**Issue by jmkiley** [mapbox/mapbox-maps-ios#62]
## Environment

- Xcode version: 12.2
- iOS version: 14.x
- Devices affected: iPhone 8, iPhone 11 
- Maps SDK Version: 10.0.0-beta.12

## Observed behavior and steps to reproduce

An error occurs when accessing a SymbolLayer that contains a value for `SymbolLayer.textField`. 

Code that reproduces the issue courtesy of @captainbarbosa.

<details><summary>Code to reproduce</summary>

```swift
// Add to viewDidLoad()
        mapView.on(.mapLoadingFinished) { [weak self] _ in
            guard let self = self else { return }
            self.updateStyle()
        }
```
```swift
func updateStyle() {
    let pointFeature = Feature(geometry: Geometry.point(Point(.mapView.centerCoordinate)))

    var geoJSONSource = GeoJSONSource()
    geoJSONSource.data = .feature(pointFeature)

    let sourceId = "my-source"
    let layerId = "my-layer"

    var symbolLayer = SymbolLayer(id: layerId)
    symbolLayer.layout?.textField = .constant("Hello world!")
    symbolLayer.source = sourceId

    _ = mapView.style.addSource(source: geoJSONSource, identifier: sourceId)
    _ = mapView.style.addLayer(layer: symbolLayer, layerPosition: nil)

    let getLayerResult = mapVC.style.getLayer(with: layerId, type: SymbolLayer.self)

    switch getLayerResult {
    case .success(let layer):
        debugPrint("✅ Got layer: \(layer)")
    case .failure(let error):
        debugPrint("❌ Got error: \(error)")
    }
```

</details>

You then hit an error in [Style.getLayer()](https://github.com/mapbox/mapbox-maps-ios/blob/82d5012201fa00b24ff13277c56563b5a4622084/Mapbox/MapboxMapsStyle/Style.swift#L142).
```

Thread 1: Fatal error: 'try!' expression unexpectedly raised an error: Error Domain=com.mapbox.bindgen Code=-2 "*** 
-[__NSDictionaryM setObject:forKey:]: object cannot be nil (key: fontStack)" UserInfo={NSUnderlyingErrorKey=*** 
-[__NSDictionaryM setObject:forKey:]: object cannot be nil (key: fontStack), NSLocalizedFailureReason=*** 
-[__NSDictionaryM setObject:forKey:]: object cannot be nil (key: fontStack)}

```

## Expected behavior

The layer is successfully returned or fails gracefully.

## Notes / preliminary analysis

There seem to be a couple of issues:
1. According to the [Mapbox Style Specification](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-text-field), `text-field` should be of type `Formatted`, whereas we have it as a `String`. We should implement `Formatted` to handle formatted strings.
2. Properties that should be optional are being required. We should figure out where this requirement is happening, then address it. While investigating, also ran into errors because there wasn't a value for `image` when retrieving a layer containing text.

## Additional links and references

cc @nishant-karajgikar

**Comment by jmkiley** [comment]
The exception itself seems to have been resolved with MapboxCoreMaps v10.0.0-beta.15. However, we should still land @nishant-karajgikar's work in [nishantk/symbolLayer-textField](https://github.com/mapbox/mapbox-maps-ios/compare/nishantk/symbolLayer-textField), which introduces the Formatted expression type.

**Comment by jmkiley** [comment]
Closing as exception is no longer happening.
